sopel/bot.py

```diff
@@ -898,6 +898,15 @@ def dispatch(self, pretrigger):
         user_obj = self.users.get(nick)
         account = user_obj.account if user_obj else None
 
+        if "time" in pretrigger.tags and pretrigger.sender in self.channels:
+            time = datetime.strptime(
+                pretrigger.tags["time"],
+                "%Y-%m-%dT%H:%M:%S.%f%z",
+            )
```

Comment:
This pattern might benefit from an additional property on the `PreTrigger` and `Trigger` objects so plugins don't have to reinvent this wheel. (That part probably wouldn't be good to put in 7.1.1, regardless of the above `%z` discussion, as it'd be an API change.)

Reply:
Bleh.

https://github.com/sopel-irc/sopel/blob/e2d8439caae33ed3dfe08f0c164d1e8323fd6f7d/sopel/trigger.py#L129-L134

Is there any reason to think stuff would break if 8.0 made that attribute into an _aware_ `datetime` in UTC?